### ISSUE TITLE
selenium side runner how to set the test speed

### ISSUE THREAD

**Issue by smithhero1994** [SeleniumHQ/selenium-ide#385]
Can you set the test speed in the selenium side runner?How can I set it?

**Comment by vnerozin** [comment]
Interesting to know too.

**Comment by tourdedave** [comment]
![recording](https://user-images.githubusercontent.com/457137/47962496-19247980-e026-11e8-8b2b-67b78acf5d53.gif)

**Comment by corevo** [comment]
He is talking about the runner, currently it isn't possible I'm afraid.  
The runner has no playback loop like the IDE so it doesn't support setting the test speed.  
I'll keep it open as a feature request.

**Comment by lievi** [comment]
There some update about that feature?

**Comment by patriciasimonep** [comment]
Is this enhancement available now?

**Comment by MisterGlass** [comment]
This feature is blocking our migrating away from selenium IDE. Since the IDE natively runs slower than the command line runner, all our tests fail when they are run in the faster CLI system.

**Comment by Fogest** [comment]
I have the problem as @MisterGlass . I developed all of the tests in the IDE and had them all working, and then I take them to the runner and 75% of them now fail. 

Is there any update on this issue?

**Comment by tudorsibiu90** [comment]
what is going on?

**Comment by tourdedave** [comment]
@MisterGlass, @Fogest We're looking to add `set speed` support in the next major version of the IDE. But, the IDE and the command-line runner both use similar synchronization strategies for all commands. So I'm not sure how using `set speed` would add additional stability to your tests in the runner, and its absence would cause a majority of your tests to break. If you could provide a reproducible test case that would be great.

A couple of things you could try in the interim:

1. Adjust the command timeout in the runner (e.g., `--timeout 30000`). It defaults to 15 seconds, whereas the IDE defaults to 30.
2. Add explicit wait commands for commands that are consistently failing ([link](https://www.seleniumhq.org/selenium-ide/docs/en/api/commands/#wait-for-element-editable))

Also, regardless of timing, there are differences between IDE and command-line playback which will crop up. Some we know about, some we don't. In the next major version of the IDE, they will all be resolved (since we'll be moving to the same run-time for both -- e.g., playback on WebDriver).

@tudorsibiu90 This is a free and open source project. We do our best to respond to all issues, forum posts, and inquiries on the project chat channel, while also working on bug fixes and new features. Your patience is appreciated.

**Comment by bhgjmorton** [comment]
Great information in this ticket, however, is there a timeline until the next major release for both to use the same run-time?

**Comment by tourdedave** [comment]
It's a major undertaking since we're fundamentally changing the runtime and the overall packaging of the application (e.g., moving from browser extension to a native app), so it's hard to say. That, and since this is an open source project, timelines and deadlines are not something that are part of our discourse. Contributions welcome though if you'd like to help out.

**Comment by jackrabbithanna** [comment]
+1 for this. Any update? 

Fell in love with Selenium IDE and then bloop can't run most tests on the command line because it executes too fast. 

This tool would be KILLER APP if could just slow down, and if the "wait for element .. " commands actually worked.

**Comment by vbisbest** [comment]
Any progress on this?  My IDE tests are also failing in side runner because of speed.   This is critical to incorporate functional tests in our DevOps CI/CD pipeline.   Hopefully this can be resolved soon.

**Comment by jawira** [comment]
I had a similar issue using `select` command, my tests were failing with _selenium-side-runner_.

First I assumed it was due to speed, but I make it work changing targets and values.

For example, this works with _Selenium IDE_ but **does not work well** with _selenium-side-runner_ (it was changing a different `<select>` element):

| Command | Target | Value |
| --- | --- | --- |
| select | id=reservation_endDate_period | label=PM |

After playing some hours, I changed _Target_ and _Value_ several times and I finally found a configuration that behaves the same in _Selenium IDE_ **and**  _selenium-side-runner_:

| Command | Target | Value |
| --- | --- | --- |
| select | name=reservation[endDate][period] | value=PM |

I also suggest to run _selenium-side-runner_ locally in **headless mode**, this way you don't have to push your changes to CI every time to try your tests. This is my `.side.yml` file:

```yaml
capabilities:
  browserName: "firefox"
  moz:firefoxOptions:
    args:
      - "-headless"
baseUrl: "http://web.project.test"
```

In conclusion, my problem wasn't due to execution speed, but instead to some Targets being inconsistent when running in headless mode. Try using different Target (Selenium IDE already proposes several suggestions) and different Values (in my case using  label=, value=, id=, index=).

I hope this will help somebody.

**Comment by rvvincelli** [comment]
I have serious doubts the CLI tool may be relied upon at the moment. We have a few nontrivial tests that work perfectly in the IDE but when run here they come out always green, no matter if some assertions are to false explicitly, no matter if headless or not.

Setting the speed is important in this sense so that we can at least see what's going on here, as the script says "all cool!" when it's not.

**Comment by JeroenAdam** [comment]
I've had the same issue and found this workaround:

add a 333ms sleep after each step by simply doing search/replace on your .side file.

search:
    }, {

replace:
    }, {        "id": "8e7b6df7-21c1-4284-8aad-efbe3ec82afd",      "comment": "",      "command": "pause",      "target": "333",      "targets": [],      "value": ""    }, {

**Comment by CharbelBs** [comment]
> @MisterGlass, @Fogest We're looking to add `set speed` support in the next major version of the IDE. But, the IDE and the command-line runner both use similar synchronization strategies for all commands. So I'm not sure how using `set speed` would add additional stability to your tests in the runner, and its absence would cause a majority of your tests to break. If you could provide a reproducible test case that would be great.
> 
> A couple of things you could try in the interim:
> 
> 1. Adjust the command timeout in the runner (e.g., `--timeout 30000`). It defaults to 15 seconds, whereas the IDE defaults to 30.
> 2. Add explicit wait commands for commands that are consistently failing ([link](https://www.seleniumhq.org/selenium-ide/docs/en/api/commands/#wait-for-element-editable))
> 
> Also, regardless of timing, there are differences between IDE and command-line playback which will crop up. Some we know about, some we don't. In the next major version of the IDE, they will all be resolved (since we'll be moving to the same run-time for both -- e.g., playback on WebDriver).
> 
> @tudorsibiu90 This is a free and open source project. We do our best to respond to all issues, forum posts, and inquiries on the project chat channel, while also working on bug fixes and new features. Your patience is appreciated.

When will the next major version of the IDE be availebal ???

**Comment by toddtarsi** [comment]
@CharbelBs - Nowadays: https://github.com/SeleniumHQ/selenium-ide/releases/tag/4.0.0-alpha.17

No speed controls though.
